sys/include/net/gnrc/nettype.h

```diff
@@ -63,6 +63,9 @@ typedef enum {
 #ifdef MODULE_GNRC_IPV6
     GNRC_NETTYPE_IPV6,          /**< Protocol is IPv6 */
 #endif
+#ifdef MODULE_GNRC_IPV6_EXT
+    GNRC_NETTYPE_IPV6_EXT,      /**< Protocol is IPv6 extension header */
```

Comment:
`gnrc_ipv6_demux`:
```
_dispatch_rcv_pkt(current->type, GNRC_NETREG_DEMUX_CTX_ALL, current);
_dispatch_rcv_pkt(GNRC_NETTYPE_IPV6, nh, current);
```

For example, if `nh == 43` (IPv6-Route), it dispatches

```
_dispatch_rcv_pkt(GNRC_NETTYPE_IPV6_EXT, GNRC_NETREG_DEMUX_CTX_ALL, current);
_dispatch_rcv_pkt(GNRC_NETTYPE_IPV6, 43, current);
```

In addition, someone may want to dispatch `GNRC_NETTYPE_IPV6_EXT` with the next header:

```
_dispatch_rcv_pkt(GNRC_NETTYPE_IPV6_EXT, 43, current);
```

I'm not sure who actually want this.